### ISSUE TITLE
Add option to allow UI text to scale with parent.

### ISSUE THREAD

**Issue by JustAnotherCodemonkey** [bevyengine/bevy#9280]
## What problem does this solve or what need does it fill?

Most games that allow for resizable windows scale their UI with the window, including text within it. In Bevy, this is currently not possible (to my knowledge).

## What solution would you like?

Optimally, one would be able to specify a font_size as a Val similar to how text works in CSS, however, to my understanding, this would be awkward as bevy_text is its own crate. Instead, a better way would be to have a set font size that works as a resolution but then also a sort of scale factor for the screen that would allow the text to stretch and shrink in order to allow what I mention above.

## What alternative(s) have you considered?

None really. I've looked around and asked the Discord and multiple people have said as it stands, this is impossible without dynamically manually changing the font size which isn't reasonable since how would I or the program know what exact size is needed to have the text fill the X dimension of the parent node?

**Comment by nicoburns** [comment]
> Optimally, one would be able to specify a font_size as a Val similar to how text works in CSS, however, to my understanding, this would be awkward as bevy_text is its own crate.

I reckon this could work. I think that `bevy_ui` could probably just translate the units into an absolute font size before passing it through to `bevy_text`.

**Comment by ickshonpe** [comment]
> > Optimally, one would be able to specify a font_size as a Val similar to how text works in CSS, however, to my understanding, this would be awkward as bevy_text is its own crate.
> 
> I reckon this could work. I think that `bevy_ui` could probably just translate the units into an absolute font size before passing it through to `bevy_text`.

Yeah, it's not very hard. Probably requires splitting `Text` into two types, one for UI and one for Text2d, which would be very desirable for a lot of other reasons anyway.

**Comment by nicoburns** [comment]
> Probably requires splitting Text into two types, one for UI and one for Text2d

If we could allow embedding UI in world-space then we might not need Text2d at all. Text2d would just become a UiText that is the child of a non-ui component (either as the UI root, or wrapped in a single UiNode)
